### ISSUE TITLE
workflow - argoexec container stuck on wait after main container finishes

### ISSUE THREAD

**Issue by m0nk3y-s3c** [argoproj/argo-workflows#7530]
## Summary

What happened/what you expected to happen?
Installed the stpes.yaml from example, but after the first step in finished the wait container get stuck and never get to step2.

What version of Argo Workflows are you running?
v3.2.6
## Diagnostics

Either a workflow that reproduces the bug, or paste you whole workflow YAML, including status, something like:

```yaml
kubectl get wf -o yaml ${workflow}
apiVersion: argoproj.io/v1alpha1
kind: Workflow
metadata:
  annotations:
    workflows.argoproj.io/pod-name-format: v1
  creationTimestamp: "2022-01-05T17:02:35Z"
  generateName: steps-
  generation: 4
  labels:
    workflows.argoproj.io/phase: Running
  name: steps-qcjw9
  namespace: argo
  resourceVersion: "3880"
  uid: 055699a2-c494-41df-9eda-62ed206661e6
spec:
  arguments:
    parameters:
    - name: message
      value: HelpMeSolveThis
  entrypoint: hello-hello-hello
  templates:
  - inputs: {}
    metadata: {}
    name: hello-hello-hello
    outputs: {}
    steps:
    - - arguments:
          parameters:
          - name: message
            value: hello1
        name: hello1
        template: whalesay
    - - arguments:
          parameters:
          - name: message
            value: hello2a
        name: hello2a
        template: whalesay
      - arguments:
          parameters:
          - name: message
            value: hello2b
        name: hello2b
        template: whalesay
  - container:
      args:
      - '{{inputs.parameters.message}}'
      command:
      - cowsay
      image: docker/whalesay
      name: ""
      resources: {}
    inputs:
      parameters:
      - name: message
    metadata: {}
    name: whalesay
    outputs: {}
status:
  artifactRepositoryRef:
    artifactRepository: {}
    default: true
  conditions:
  - status: "True"
    type: PodRunning
  finishedAt: null
  nodes:
    steps-qcjw9:
      children:
      - steps-qcjw9-3301525239
      displayName: steps-qcjw9
      finishedAt: null
      id: steps-qcjw9
      name: steps-qcjw9
      phase: Running
      progress: 0/1
      startedAt: "2022-01-05T17:02:35Z"
      templateName: hello-hello-hello
      templateScope: local/steps-qcjw9
      type: Steps
    steps-qcjw9-2094437448:
      boundaryID: steps-qcjw9
      displayName: hello1
      finishedAt: null
      hostNodeName: gke-bug-bounty-argo-default-pool-6465c554-06m1
      id: steps-qcjw9-2094437448
      inputs:
        parameters:
        - name: message
          value: hello1
      name: steps-qcjw9[0].hello1
      phase: Running
      progress: 0/1
      startedAt: "2022-01-05T17:02:35Z"
      templateName: whalesay
      templateScope: local/steps-qcjw9
      type: Pod
    steps-qcjw9-3301525239:
      boundaryID: steps-qcjw9
      children:
      - steps-qcjw9-2094437448
      displayName: '[0]'
      finishedAt: null
      id: steps-qcjw9-3301525239
      name: steps-qcjw9[0]
      phase: Running
      progress: 0/1
      startedAt: "2022-01-05T17:02:35Z"
      templateScope: local/steps-qcjw9
      type: StepGroup
  phase: Running
  progress: 0/1
  startedAt: "2022-01-05T17:02:35Z"
```

What Kubernetes provider are you using?
GKE
What executor are you running? Docker/K8SAPI/Kubelet/PNS/Emissary
No Idea, this is my installation steps:
kubectl create namespace argo
kubectl apply -n argo -f https://github.com/argoproj/argo-workflows/releases/download/v3.2.6/install.yaml
curl -sLO https://github.com/argoproj/argo-workflows/releases/download/v3.2.6/argo-linux-amd64.gz ... installing client
argo -n argo submit ./steps.yaml -p message="HelpMeSolveThis"

```bash
# Logs from the workflow controller:
kubectl logs -n argo deploy/workflow-controller | grep ${workflow}
time="2022-01-05T16:59:10.089Z" level=info msg="Processing workflow" namespace=default workflow=steps-z4csn
time="2022-01-05T16:59:10.103Z" level=info msg="Updated phase  -> Running" namespace=default workflow=steps-z4csn
time="2022-01-05T16:59:10.104Z" level=info msg="Steps node steps-z4csn initialized Running" namespace=default workflow=steps-z4csn
time="2022-01-05T16:59:10.104Z" level=info msg="StepGroup node steps-z4csn-2699925059 initialized Running" namespace=default workflow=steps-z4csn
time="2022-01-05T16:59:10.105Z" level=info msg="Pod node steps-z4csn-3230619924 initialized Pending" namespace=default workflow=steps-z4csn
time="2022-01-05T16:59:10.177Z" level=info msg="Created pod: steps-z4csn[0].hello1 (steps-z4csn-3230619924)" namespace=default workflow=steps-z4csn
time="2022-01-05T16:59:10.177Z" level=info msg="Workflow step group node steps-z4csn-2699925059 not yet completed" namespace=default workflow=steps-z4csn
time="2022-01-05T16:59:10.177Z" level=info msg="TaskSet Reconciliation" namespace=default workflow=steps-z4csn
time="2022-01-05T16:59:10.177Z" level=info msg=reconcileAgentPod namespace=default workflow=steps-z4csn
time="2022-01-05T16:59:10.239Z" level=info msg="Workflow update successful" namespace=default phase=Running resourceVersion=1993 workflow=steps-z4csn
time="2022-01-05T16:59:20.190Z" level=info msg="Processing workflow" namespace=default workflow=steps-z4csn
time="2022-01-05T16:59:20.191Z" level=info msg="Updating node steps-z4csn-3230619924 message: ContainerCreating" namespace=default workflow=steps-z4csn
time="2022-01-05T16:59:20.192Z" level=info msg="Workflow step group node steps-z4csn-2699925059 not yet completed" namespace=default workflow=steps-z4csn
time="2022-01-05T16:59:20.192Z" level=info msg="TaskSet Reconciliation" namespace=default workflow=steps-z4csn
time="2022-01-05T16:59:20.192Z" level=info msg=reconcileAgentPod namespace=default workflow=steps-z4csn
time="2022-01-05T16:59:20.208Z" level=info msg="Workflow update successful" namespace=default phase=Running resourceVersion=2139 workflow=steps-z4csn
time="2022-01-05T16:59:30.211Z" level=info msg="Processing workflow" namespace=default workflow=steps-z4csn
time="2022-01-05T16:59:30.213Z" level=info msg="Workflow step group node steps-z4csn-2699925059 not yet completed" namespace=default workflow=steps-z4csn
time="2022-01-05T16:59:30.213Z" level=info msg="TaskSet Reconciliation" namespace=default workflow=steps-z4csn
time="2022-01-05T16:59:30.213Z" level=info msg=reconcileAgentPod namespace=default workflow=steps-z4csn
time="2022-01-05T16:59:52.122Z" level=info msg="Processing workflow" namespace=default workflow=steps-z4csn
time="2022-01-05T16:59:52.122Z" level=info msg="Updating node steps-z4csn-3230619924 status Pending -> Running" namespace=default workflow=steps-z4csn
time="2022-01-05T16:59:52.123Z" level=info msg="Workflow step group node steps-z4csn-2699925059 not yet completed" namespace=default workflow=steps-z4csn
time="2022-01-05T16:59:52.123Z" level=info msg="TaskSet Reconciliation" namespace=default workflow=steps-z4csn
time="2022-01-05T16:59:52.123Z" level=info msg=reconcileAgentPod namespace=default workflow=steps-z4csn
time="2022-01-05T16:59:52.143Z" level=info msg="Workflow update successful" namespace=default phase=Running resourceVersion=2384 workflow=steps-z4csn
time="2022-01-05T17:00:02.142Z" level=info msg="Processing workflow" namespace=default workflow=steps-z4csn
time="2022-01-05T17:00:02.143Z" level=info msg="Workflow step group node steps-z4csn-2699925059 not yet completed" namespace=default workflow=steps-z4csn
time="2022-01-05T17:00:02.143Z" level=info msg="TaskSet Reconciliation" namespace=default workflow=steps-z4csn
time="2022-01-05T17:00:02.143Z" level=info msg=reconcileAgentPod namespace=default workflow=steps-z4csn
time="2022-01-05T17:02:35.345Z" level=info msg="Processing workflow" namespace=argo workflow=steps-qcjw9
time="2022-01-05T17:02:35.362Z" level=info msg="Updated phase  -> Running" namespace=argo workflow=steps-qcjw9
time="2022-01-05T17:02:35.363Z" level=info msg="Steps node steps-qcjw9 initialized Running" namespace=argo workflow=steps-qcjw9
time="2022-01-05T17:02:35.364Z" level=info msg="StepGroup node steps-qcjw9-3301525239 initialized Running" namespace=argo workflow=steps-qcjw9
time="2022-01-05T17:02:35.364Z" level=info msg="Pod node steps-qcjw9-2094437448 initialized Pending" namespace=argo workflow=steps-qcjw9
time="2022-01-05T17:02:35.402Z" level=info msg="Created pod: steps-qcjw9[0].hello1 (steps-qcjw9-2094437448)" namespace=argo workflow=steps-qcjw9
time="2022-01-05T17:02:35.402Z" level=info msg="Workflow step group node steps-qcjw9-3301525239 not yet completed" namespace=argo workflow=steps-qcjw9
time="2022-01-05T17:02:35.402Z" level=info msg="TaskSet Reconciliation" namespace=argo workflow=steps-qcjw9
time="2022-01-05T17:02:35.403Z" level=info msg=reconcileAgentPod namespace=argo workflow=steps-qcjw9
time="2022-01-05T17:02:35.424Z" level=info msg="Workflow update successful" namespace=argo phase=Running resourceVersion=3577 workflow=steps-qcjw9
time="2022-01-05T17:02:45.416Z" level=info msg="Processing workflow" namespace=argo workflow=steps-qcjw9
time="2022-01-05T17:02:45.417Z" level=info msg="Updating node steps-qcjw9-2094437448 message: ContainerCreating" namespace=argo workflow=steps-qcjw9
time="2022-01-05T17:02:45.418Z" level=info msg="Workflow step group node steps-qcjw9-3301525239 not yet completed" namespace=argo workflow=steps-qcjw9
time="2022-01-05T17:02:45.418Z" level=info msg="TaskSet Reconciliation" namespace=argo workflow=steps-qcjw9
time="2022-01-05T17:02:45.418Z" level=info msg=reconcileAgentPod namespace=argo workflow=steps-qcjw9
time="2022-01-05T17:02:45.443Z" level=info msg="Workflow update successful" namespace=argo phase=Running resourceVersion=3652 workflow=steps-qcjw9
time="2022-01-05T17:02:55.445Z" level=info msg="Processing workflow" namespace=argo workflow=steps-qcjw9
time="2022-01-05T17:02:55.447Z" level=info msg="Workflow step group node steps-qcjw9-3301525239 not yet completed" namespace=argo workflow=steps-qcjw9
time="2022-01-05T17:02:55.447Z" level=info msg="TaskSet Reconciliation" namespace=argo workflow=steps-qcjw9
time="2022-01-05T17:02:55.447Z" level=info msg=reconcileAgentPod namespace=argo workflow=steps-qcjw9
time="2022-01-05T17:03:15.662Z" level=info msg="Processing workflow" namespace=argo workflow=steps-qcjw9
time="2022-01-05T17:03:15.663Z" level=info msg="Updating node steps-qcjw9-2094437448 status Pending -> Running" namespace=argo workflow=steps-qcjw9
time="2022-01-05T17:03:15.665Z" level=info msg="Workflow step group node steps-qcjw9-3301525239 not yet completed" namespace=argo workflow=steps-qcjw9
time="2022-01-05T17:03:15.665Z" level=info msg="TaskSet Reconciliation" namespace=argo workflow=steps-qcjw9
time="2022-01-05T17:03:15.665Z" level=info msg=reconcileAgentPod namespace=argo workflow=steps-qcjw9
time="2022-01-05T17:03:15.689Z" level=info msg="Workflow update successful" namespace=argo phase=Running resourceVersion=3880 workflow=steps-qcjw9
time="2022-01-05T17:03:25.690Z" level=info msg="Processing workflow" namespace=argo workflow=steps-qcjw9
time="2022-01-05T17:03:25.692Z" level=info msg="Workflow step group node steps-qcjw9-3301525239 not yet completed" namespace=argo workflow=steps-qcjw9
time="2022-01-05T17:03:25.692Z" level=info msg="TaskSet Reconciliation" namespace=argo workflow=steps-qcjw9
time="2022-01-05T17:03:25.692Z" level=info msg=reconcileAgentPod namespace=argo workflow=steps-qcjw9
time="2022-01-05T17:23:25.691Z" level=info msg="Processing workflow" namespace=argo workflow=steps-qcjw9
time="2022-01-05T17:23:25.693Z" level=info msg="Workflow step group node steps-qcjw9-3301525239 not yet completed" namespace=argo workflow=steps-qcjw9
time="2022-01-05T17:23:25.693Z" level=info msg="TaskSet Reconciliation" namespace=argo workflow=steps-qcjw9
time="2022-01-05T17:23:25.693Z" level=info msg=reconcileAgentPod namespace=argo workflow=steps-qcjw9
time="2022-01-05T17:43:25.692Z" level=info msg="Processing workflow" namespace=argo workflow=steps-qcjw9
time="2022-01-05T17:43:25.694Z" level=info msg="Workflow step group node steps-qcjw9-3301525239 not yet completed" namespace=argo workflow=steps-qcjw9
time="2022-01-05T17:43:25.694Z" level=info msg="TaskSet Reconciliation" namespace=argo workflow=steps-qcjw9
time="2022-01-05T17:43:25.694Z" level=info msg=reconcileAgentPod namespace=argo workflow=steps-qcjw9

this also might help "logs from wait container"
![image](https://user-images.githubusercontent.com/37670762/148259631-bd3c3cfb-31f8-475c-a9f2-23dbd88cc882.png)



# If the workflow's pods have not been created, you can skip the rest of the diagnostics.

# The workflow's pods that are problematic:
kubectl get pod -o yaml -l workflows.argoproj.io/workflow=${workflow},workflow.argoproj.io/phase!=Succeeded

# Logs from in your workflow's wait container, something like:
kubectl logs -c wait -l workflows.argoproj.io/workflow=${workflow},workflow.argoproj.io/phase!=Succeeded
```

---

**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritise the issues with the most 👍.

**Comment by sarabala1979** [comment]
@m0nk3y-s3c can you provide the wait container log?

**Comment by sarabala1979** [comment]
![image](https://user-images.githubusercontent.com/33908564/148715383-0f73b322-ae47-4b28-aa53-1f9e6c0ce26e.png)
looks like  you using  `docker`.

Can you try `PNS`?
you can  configure `containerRuntimeExecutor: pns`  in configmap `workflow-controller-configmap`

**Comment by TanmayKedari** [comment]
> ![image](https://user-images.githubusercontent.com/33908564/148715383-0f73b322-ae47-4b28-aa53-1f9e6c0ce26e.png) looks like you using `docker`.
> 
> Can you try `PNS`? you can configure `containerRuntimeExecutor: pns` in configmap `workflow-controller-configmap`

Facing same issue while using k8sapi executor
Tried using pns no luck there as well

**Comment by sarabala1979** [comment]
@TanmayKedari can you provide the wait container log?  are you using  v3.2.6?

**Comment by TanmayKedari** [comment]
time="2022-01-10T05:56:13.016Z" level=info msg="Starting Workflow Executor" executorType=k8sapi version=v3.1.0
time="2022-01-10T05:56:13.027Z" level=info msg="Creating a K8sAPI executor"
time="2022-01-10T05:56:13.027Z" level=info msg="Executor initialized" includeScriptOutput=false namespace=ds-argo podName=ftl004cp-lpxqv-1641788760-3356570892 template="{\"name\":\"adapter-etl\",\"inputs\":{\"parameters\":[{\"name\":\"jobname\",\"value\":\"FTL004CP\"}]},\"outputs\":{},\"affinity\":{\"nodeAffinity\":{\"requiredDuringSchedulingIgnoredDuringExecution\":{\"nodeSelectorTerms\":[{\"matchExpressions\":[{\"key\":\"biz\",\"operator\":\"In\",\"values\":[\"argo\"]},{\"key\":\"layer\",\"operator\":\"In\",\"values\":[\"app\"]},{\"key\":\"app\",\"operator\":\"In\",\"values\":[\"all\"]},{\"key\":\"env\",\"operator\":\"In\",\"values\":[\"prod\"]},{\"key\":\"kubernetes.io/hostname\",\"operator\":\"In\",\"values\":[\"aks-epk8sargo-39501614-vmss00000d\"]}]}]}}},\"metadata\":{},\"container\":{\"name\":\"adapter\",\"image\":\"yard.elasticrun.in/adapter-etl:release\",\"env\":[{\"name\":\"logLevel\",\"value\":\"Basic\"},{\"name\":\"etl\",\"value\":\"adapter-etl\"},{\"name\":\"job\",\"value\":\"FTL004CP\"}],\"resources\":{\"limits\":{\"cpu\":\"1500m\"},\"requests\":{\"cpu\":\"100m\"}},\"volumeMounts\":[{\"name\":\"adapter-etl-vol\",\"mountPath\":\"/ntex/application/sharedvols\"},{\"name\":\"adapter-etl-logs\",\"mountPath\":\"/ntex/application/sharedvols/adapter-etl/log\"},{\"name\":\"adapter-etl-kettle-properties-vol\",\"readOnly\":true,\"mountPath\":\"/ntex/application/app/adapter-etl/kettle-home/.kettle/kettle.properties\",\"subPath\":\"kettle.properties\"},{\"name\":\"adapter-etl-repositories-xml-vol\",\"readOnly\":true,\"mountPath\":\"/ntex/application/app/adapter-etl/kettle-home/.kettle/repositories.xml\",\"subPath\":\"repositories.xml\"},{\"name\":\"adapter-etl-shared-xml-vol\",\"readOnly\":true,\"mountPath\":\"/ntex/application/app/adapter-etl/kettle-home/.kettle/shared.xml\",\"subPath\":\"shared.xml\"},{\"name\":\"adapter-etl-job-properties-vol\",\"readOnly\":true,\"mountPath\":\"/ntex/application/app/adapter-etl/kettle-base/conf/job.properties\",\"subPath\":\"job.properties\"},{\"name\":\"adapter-etl-dates-csv-vol\",\"readOnly\":true,\"mountPath\":\"/ntex/application/app/adapter-etl/kettle-base/conf/dates.csv\",\"subPath\":\"dates.csv\"}],\"imagePullPolicy\":\"Always\"},\"volumes\":[{\"name\":\"adapter-etl-vol\",\"persistentVolumeClaim\":{\"claimName\":\"er-engg-ds-argo-adapter-pvc\"}},{\"name\":\"adapter-etl-logs\",\"persistentVolumeClaim\":{\"claimName\":\"er-engg-ds-argo-pvc\"}},{\"name\":\"adapter-etl-kettle-properties-vol\",\"configMap\":{\"name\":\"adapter-etl-kettle-properties-config\",\"items\":[{\"key\":\"kettle.properties\",\"path\":\"kettle.properties\"}]}},{\"name\":\"adapter-etl-job-properties-vol\",\"configMap\":{\"name\":\"adapter-etl-job-properties-bjob06-config\",\"items\":[{\"key\":\"job.properties\",\"path\":\"job.properties\"}]}},{\"name\":\"adapter-etl-repositories-xml-vol\",\"configMap\":{\"name\":\"adapter-etl-repositories-xml-config\",\"items\":[{\"key\":\"repositories.xml\",\"path\":\"repositories.xml\"}]}},{\"name\":\"adapter-etl-shared-xml-vol\",\"configMap\":{\"name\":\"adapter-etl-shared-xml-config\",\"items\":[{\"key\":\"shared.xml\",\"path\":\"shared.xml\"}]}},{\"name\":\"adapter-etl-dates-csv-vol\",\"configMap\":{\"name\":\"adapter-etl-dates-csv-config\",\"items\":[{\"key\":\"dates.csv\",\"path\":\"dates.csv\"}]}},{\"name\":\"adapter-etl-fluentd\",\"configMap\":{\"name\":\"adapter-etl-fluentd-config\",\"items\":[{\"key\":\"fluent.conf\",\"path\":\"fluent.conf\"}]}}],\"sidecars\":[{\"name\":\"fluentd\",\"image\":\"yard.elasticrun.in/er-fluentd-base:release\",\"resources\":{},\"volumeMounts\":[{\"name\":\"adapter-etl-fluentd\",\"mountPath\":\"/fluentd/etc\"},{\"name\":\"adapter-etl-logs\",\"mountPath\":\"/fluentd/log\"}]}],\"archiveLocation\":{\"archiveLogs\":true,\"s3\":{\"endpoint\":\"minio:9000\",\"bucket\":\"my-bucket\",\"insecure\":true,\"accessKeySecret\":{\"name\":\"my-minio-cred\",\"key\":\"accesskey\"},\"secretKeySecret\":{\"name\":\"my-minio-cred\",\"key\":\"secretkey\"},\"key\":\"ftl004cp-lpxqv-1641788760/ftl004cp-lpxqv-1641788760-3356570892\"}},\"hostAliases\":[{\"ip\":\"10.210.0.4\",\"hostnames\":[\"web.ntex.com\"]},{\"ip\":\"10.210.0.5\",\"hostnames\":[\"admin.ntex.com\"]},{\"ip\":\"10.100.0.17\",\"hostnames\":[\"swarm.ntex.com\"]},{\"ip\":\"10.210.2.5\",\"hostnames\":[\"ifsc.ntex.com\"]},{\"ip\":\"10.171.2.66\",\"hostnames\":[\"cas.ntex.com\"]},{\"ip\":\"10.210.2.8\",\"hostnames\":[\"beejakf.ntex.com\",\"beejakn.ntex.com\",\"frappe.ntex.com\",\"beejak.ntex.com\"]},{\"ip\":\"10.210.2.17\",\"hostnames\":[\"lswarm.ntex.com\"]},{\"ip\":\"10.210.2.23\",\"hostnames\":[\"recon.ntex.com\"]},{\"ip\":\"10.210.2.25\",\"hostnames\":[\"jobs.ntex.com\"]},{\"ip\":\"10.210.2.21\",\"hostnames\":[\"spine.ntex.com\"]},{\"ip\":\"10.210.2.13\",\"hostnames\":[\"insights.ntex.com\"]},{\"ip\":\"10.200.150.151\",\"hostnames\":[\"elk.ntex.com\"]},{\"ip\":\"10.50.0.53\",\"hostnames\":[\"bswarm.ntex.com\"]},{\"ip\":\"10.210.2.19\",\"hostnames\":[\"buddhi.ntex.com\"]},{\"ip\":\"10.210.2.14\",\"hostnames\":[\"libera.ntex.com\"]},{\"ip\":\"10.210.2.24\",\"hostnames\":[\"xplorun.ntex.com\"]},{\"ip\":\"127.0.0.1\",\"hostnames\":[\"app.ntex.com\"]},{\"ip\":\"10.210.4.5\",\"hostnames\":[\"mongodb.ntex.com\",\"mongodb1.ntex.com\"]},{\"ip\":\"10.210.4.7\",\"hostnames\":[\"mongodb-arb.ntex.com\",\"mariadb.ntex.com\",\"insightsdb.ntex.com\"]},{\"ip\":\"10.210.4.4\",\"hostnames\":[\"pgsql.ntex.com\"]}]}" version="&Version{Version:v3.1.0,BuildDate:2021-06-21T22:37:12Z,GitCommit:caeb46d443b6491fdd9716c6815ab29525eac632,GitTag:v3.1.0,GitTreeState:clean,GoVersion:go1.15.7,Compiler:gc,Platform:linux/amd64,}"
time="2022-01-10T05:56:13.027Z" level=info msg="Starting annotations monitor"
time="2022-01-10T05:56:13.028Z" level=info msg="Starting deadline monitor"
time="2022-01-10T05:56:13.050Z" level=info msg="Watch pods 200"
time="2022-01-10T06:01:13.027Z" level=info msg="Alloc=4422 TotalAlloc=11140 Sys=73041 NumGC=6 Goroutines=10"
time="2022-01-10T06:06:13.027Z" level=info msg="Alloc=4380 TotalAlloc=11155 Sys=73041 NumGC=8 Goroutines=10"
time="2022-01-10T06:11:13.027Z" level=info msg="Alloc=4377 TotalAlloc=11170 Sys=73041 NumGC=11 Goroutines=10"
time="2022-01-10T06:16:13.027Z" level=info msg="Alloc=4379 TotalAlloc=11185 Sys=73041 NumGC=13 Goroutines=10"
time="2022-01-10T06:21:13.027Z" level=info msg="Alloc=4379 TotalAlloc=11201 Sys=73041 NumGC=16 Goroutines=10"
time="2022-01-10T06:26:13.027Z" level=info msg="Alloc=4382 TotalAlloc=11217 Sys=73041 NumGC=18 Goroutines=10"
time="2022-01-10T06:31:13.027Z" level=info msg="Alloc=4381 TotalAlloc=11233 Sys=73041 NumGC=21 Goroutines=10"
time="2022-01-10T06:36:13.027Z" level=info msg="Alloc=4385 TotalAlloc=11249 Sys=73041 NumGC=23 Goroutines=10"
time="2022-01-10T06:41:13.029Z" level=info msg="Alloc=4383 TotalAlloc=11264 Sys=73041 NumGC=26 Goroutines=10"
time="2022-01-10T06:41:34.062Z" level=info msg="Watch pods 200"
time="2022-01-10T06:46:13.027Z" level=info msg="Alloc=4388 TotalAlloc=11659 Sys=73041 NumGC=28 Goroutines=10"
time="2022-01-10T06:51:13.027Z" level=info msg="Alloc=4387 TotalAlloc=11676 Sys=73041 NumGC=31 Goroutines=10"
time="2022-01-10T06:56:13.027Z" level=info msg="Alloc=4390 TotalAlloc=11692 Sys=73041 NumGC=33 Goroutines=10"
time="2022-01-10T07:01:13.027Z" level=info msg="Alloc=4388 TotalAlloc=11707 Sys=73041 NumGC=36 Goroutines=10"
time="2022-01-10T07:06:13.027Z" level=info msg="Alloc=4390 TotalAlloc=11721 Sys=73041 NumGC=38 Goroutines=10"
time="2022-01-10T07:11:13.027Z" level=info msg="Alloc=4388 TotalAlloc=11736 Sys=73041 NumGC=41 Goroutines=10"

**Comment by TanmayKedari** [comment]
time="2022-01-10T05:56:13.016Z" level=info msg="Starting Workflow Executor" executorType=k8sapi version=v3.1.0
time="2022-01-10T05:56:13.027Z" level=info msg="Creating a K8sAPI executor"
time="2022-01-10T05:56:13.027Z" level=info msg="Executor initialized" includeScriptOutput=false namespace=ds-argo  version="&Version{Version:v3.1.0,BuildDate:2021-06-21T22:37:12Z,GitCommit:caeb46d443b6491fdd9716c6815ab29525eac632,GitTag:v3.1.0,GitTreeState:clean,GoVersion:go1.15.7,Compiler:gc,Platform:linux/amd64,}"
time="2022-01-10T05:56:13.027Z" level=info msg="Starting annotations monitor"
time="2022-01-10T05:56:13.028Z" level=info msg="Starting deadline monitor"
time="2022-01-10T05:56:13.050Z" level=info msg="Watch pods 200"
time="2022-01-10T06:01:13.027Z" level=info msg="Alloc=4422 TotalAlloc=11140 Sys=73041 NumGC=6 Goroutines=10"
time="2022-01-10T06:06:13.027Z" level=info msg="Alloc=4380 TotalAlloc=11155 Sys=73041 NumGC=8 Goroutines=10"
time="2022-01-10T06:11:13.027Z" level=info msg="Alloc=4377 TotalAlloc=11170 Sys=73041 NumGC=11 Goroutines=10"
time="2022-01-10T06:16:13.027Z" level=info msg="Alloc=4379 TotalAlloc=11185 Sys=73041 NumGC=13 Goroutines=10"
time="2022-01-10T06:21:13.027Z" level=info msg="Alloc=4379 TotalAlloc=11201 Sys=73041 NumGC=16 Goroutines=10"
time="2022-01-10T06:26:13.027Z" level=info msg="Alloc=4382 TotalAlloc=11217 Sys=73041 NumGC=18 Goroutines=10"
time="2022-01-10T06:31:13.027Z" level=info msg="Alloc=4381 TotalAlloc=11233 Sys=73041 NumGC=21 Goroutines=10"
time="2022-01-10T06:36:13.027Z" level=info msg="Alloc=4385 TotalAlloc=11249 Sys=73041 NumGC=23 Goroutines=10"
time="2022-01-10T06:41:13.029Z" level=info msg="Alloc=4383 TotalAlloc=11264 Sys=73041 NumGC=26 Goroutines=10"
time="2022-01-10T06:41:34.062Z" level=info msg="Watch pods 200"
time="2022-01-10T06:46:13.027Z" level=info msg="Alloc=4388 TotalAlloc=11659 Sys=73041 NumGC=28 Goroutines=10"
time="2022-01-10T06:51:13.027Z" level=info msg="Alloc=4387 TotalAlloc=11676 Sys=73041 NumGC=31 Goroutines=10"
time="2022-01-10T06:56:13.027Z" level=info msg="Alloc=4390 TotalAlloc=11692 Sys=73041 NumGC=33 Goroutines=10"
time="2022-01-10T07:01:13.027Z" level=info msg="Alloc=4388 TotalAlloc=11707 Sys=73041 NumGC=36 Goroutines=10"
time="2022-01-10T07:06:13.027Z" level=info msg="Alloc=4390 TotalAlloc=11721 Sys=73041 NumGC=38 Goroutines=10"
time="2022-01-10T07:11:13.027Z" level=info msg="Alloc=4388 TotalAlloc=11736 Sys=73041 NumGC=41 Goroutines=10"

**Comment by sarabala1979** [comment]
@TanmayKedari  you using old release `version=v3.1.0` Please upgrade to 3.1.16 or v3.2.6

**Comment by sarabala1979** [comment]
@TanmayKedari Is your container finishing quickly?

**Comment by TanmayKedari** [comment]
> @TanmayKedari you using old release `version=v3.1.0` Please upgrade to 3.1.16 or v3.2.6

Was facing the issue previously on version=v3.0.3 since have upgraded to v3.1.0.
is this issue resolved in 3.1.16?

**Comment by sarabala1979** [comment]
There is few improvements and fix’s  on 3.1.16 but I would recommend to use latest release v3.2.6

**Comment by TanmayKedari** [comment]
> @TanmayKedari Is your container finishing quickly?

One container ie main finishes properly after proper execution and argoexec does not finish.
Not sure about which container you are referring to but I hope this answers your question.

P.S - The Same issue is not observed in Sample pre-written workflows which are available during submit new workflow

**Comment by TanmayKedari** [comment]
> There is few improvements and fix’s on 3.1.16 but I would recommend to use latest release v3.2.6
> […](#)
> Sent from my iPhone
> On Jan 9, 2022, at 11:29 PM, Tanmay Kedari ***@***.***> wrote: ﻿ @TanmayKedari you using old release version=v3.1.0 Please upgrade to 3.1.16 or v3.2.6 Was facing the issue previously on version=v3.0.3 since have upgraded to v3.1.0. is this issue resolved in 3.1.16? — Reply to this email directly, view it on GitHub, or unsubscribe. Triage notifications on the go with GitHub Mobile for iOS or Android. You are receiving this because you commented.

It worked 👍👍🥳🥳
Thanks a lot 
The execution was properly Finished.

**Comment by alexec** [comment]
Can you please try the latest version?
